packages/camera/camera_web/lib/src/types/camera_error_code.dart

```diff
@@ -32,7 +32,7 @@ class CameraErrorCode {
   /// The camera cannot be used or the permission
   /// to access the camera is not granted.
   static const CameraErrorCode permissionDenied =
-      CameraErrorCode._('cameraPermission');
+      CameraErrorCode._('CameraAccessDenied');
 
   /// The camera options are incorrect or attempted
   /// to access the media input from an insecure context.
```

Comment:
There's many other error codes in this file. Would it make sense to go through all of them so we don't have to do multiple breaking changes like this in the future?

I'm also guessing that this plugin is using this class to eventually throw a `CameraException` back to the plugin ([like here](https://github.com/flutter/plugins/blob/4479e26172b997a4118599ca45742a7ef224e5b6/packages/camera/camera_web/lib/src/camera_web.dart#L188-L194)).

Wouldn't it make more sense to have different types of `CameraException` [here](https://github.com/flutter/plugins/blob/main/packages/camera/camera_platform_interface/lib/src/types/camera_exception.dart), that can be reused across platforms rather than attempting to standardize "Magic Strings" across packages?

/cc @stuartmorgan

Reply:
@ditman I agree that having multiple breaking changes is bad dev experience. 

The permission exceptions "feels" like a special case to me because iOS did not have it before, and it's added to fix a crash on iOS. So I'd prefer having this API change sooner. Otherwise developers may have to use legacy permission exceptions for a long time (like this code in README): 
```
try { controller.initialize() } 
catch (CameraAccessRestricted) { ... } // iOS only
catch (CameraAccessDeniedWithoutPrompt) { ... } // iOS only
catch (cameraPermission) { ... } // legacy, used in web only. 
```
And in the future after we fix the web API, they will have change this permission handling code again. This is also not a great developer experience. But I am fine either way.

Reply:
> Wouldn't it make more sense to have different types of `CameraException` [here](https://github.com/flutter/plugins/blob/main/packages/camera/camera_platform_interface/lib/src/types/camera_exception.dart), that can be reused across platforms rather than attempting to standardize "Magic Strings" across packages?

The goal is to to move to constants for the codes, so the strings aren't magic: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#platform-exception-handling

Reply:
Hi @ditman and @stuartmorgan I am following up if you are fine with me landing it? or is there any issue with my previous comment?